runtime/src/commands.rs

```diff
@@ -0,0 +1,35 @@
+use caliptra_common::cprintln;
+use caliptra_drivers::{CaliptraResult, Mailbox};
+use caliptra_registers::mbox;
+use caliptra_registers::mbox::enums::MboxStatusE;
+
+fn wait_for_cmd() {
+    #[cfg(feature = "riscv")]
+    unsafe {
+        // wfi causes an exception. Need to investigate.
+        //core::arch::asm!("wfi");
+    }
+}
+
+pub fn handle_mailbox_commands() -> CaliptraResult<()> {
+    loop {
+        wait_for_cmd();
+
+        if let Some(mut recv_txn) = Mailbox::default().try_start_recv_txn() {
+            let cmd_id = recv_txn.cmd();
+            let len = recv_txn.dlen() as usize;
+            let mut buf = [0u32; 1024];
+            recv_txn.copy_request(0, &mut buf[..len]).unwrap();
+            recv_txn.complete(MboxStatusE::CmdBusy);
+
+            // TODO: Actually handle command
+            cprintln!("[rt] Received command={}, len={}", cmd_id, len);
+
+            // Write response
+            let out_buf = [0xFFu8; 16];
+            let mut send_txn = Mailbox::default().try_start_send_txn().unwrap();
```

Comment:
You don't want to use try_start_send_txn to send a response; that's used for making an "RPC" request to the SoC.